planner/core/plan.go

```diff
@@ -166,12 +166,15 @@ type LogicalPlan interface {
 	pushDownTopN(topN *LogicalTopN) LogicalPlan
 
 	// recursiveDeriveStats derives statistic info between plans.
-	recursiveDeriveStats() (*property.StatsInfo, error)
+	recursiveDeriveStats(colGroups [][]*expression.Column) (*property.StatsInfo, error)
 
 	// DeriveStats derives statistic info for current plan node given child stats.
 	// We need selfSchema, childSchema here because it makes this method can be used in
 	// cascades planner, where LogicalPlan might not record its children or schema.
-	DeriveStats(childStats []*property.StatsInfo, selfSchema *expression.Schema, childSchema []*expression.Schema) (*property.StatsInfo, error)
+	DeriveStats(childStats []*property.StatsInfo, selfSchema *expression.Schema, childSchema []*expression.Schema, colGroups [][]*expression.Column) (*property.StatsInfo, error)
+
+	// ExtractColGroups extracts column groups whose DNV are required by this operator from child.
+	ExtractColGroups(colGroups [][]*expression.Column) [][]*expression.Column
```

Comment:
Could you please add more comments to this method? It's better to add some examples...